### PR TITLE
Consolidate API calls to TaskContext

### PR DESCRIPTION
**Pull request by AbdBarho**
code clean up, delegate all API calls to hook, so that the react code can focus on the data and representation.

Many of the changes is just shuffling interfaces around.

I want to make a couple of PRs after this one to continue cleaning up the code, I am not sure if the `TaskContext` as a provider is the ultimate solution for the long run, but it made sense for now.

### CODE REVIEW

**Review by fozziethebeat — APPROVED** [review]

**Comment by fozziethebeat** [comment]
Overall looks like a good cleanup once it builds right.

**Comment by AbdBarho** [comment]
@fozziethebeat I made significant changes in the last commit, I will re-request the review.

**Comment by AbdBarho** [comment]
@othrayte you can also review, this touches many of your recent changes.

**Review by 53jk1 — APPROVED** [review]

File: website/src/types/Hooks.ts [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-}
-
-export type TaskApiHook<Task extends BaseTask> = {
+export type TaskApiHook<Task extends BaseTask, ResponseContent> = {
   response: TaskResponse<Task>;
   isLoading: boolean;
-  completeTask: (interaction: TaskInteraction) => void;
-  skipTask: () => void;
+  completeTask: (interaction: ResponseContent) => Promise<void>;
+  skipTask: () => Promise<void>;
```
Comment (othrayte) [inline website/src/types/Hooks.ts:7]: I don't think we need skipTask if we have rejectTask. Probably best not to expose a way to accidentally skip without informing the server we are rejecting.

**Review by othrayte — APPROVED** [review]
Great to see some refactoring. 👷

File: website/src/types/Hooks.ts (hunk 1 above)
Comment (AbdBarho) [reply website/src/types/Hooks.ts:7]: Agree, I will remove it in the next PR.

**Pull request merged**